src/scripts/main.js

```diff
@@ -1 +1,88 @@
 'use strict';
+
+// #region  Promise 1
+const promise1 = new Promise((resolve, reject) => {
+  let clicked = false;
+
+  document.addEventListener('click' || 'contextmenu', (e) => {
+    clicked = true;
+    resolve('First promise was resolved');
+  });
+
+  if (!clicked) {
+    setTimeout(() => {
+      reject(new Error('First promise was rejected'));
+    }, 3000);
+  }
```

Comment:
clicked variable is redudant
```suggestion
  document.addEventListener('click', (e) => {
    resolve('First promise was resolved');
  });

    setTimeout(() => {
      reject(new Error('First promise was rejected'));
    }, 3000);
```